src/projections/fouc_s.cpp

```diff
@@ -33,22 +33,22 @@ static PJ_XY fouc_s_s_forward (PJ_LP lp, PJ *P) {           /* Spheroidal, forwa
 static PJ_LP fouc_s_s_inverse (PJ_XY xy, PJ *P) {           /* Spheroidal, inverse */
     PJ_LP lp = {0.0,0.0};
     struct pj_opaque *Q = static_cast<struct pj_opaque*>(P->opaque);
-    double V;
     int i;
 
     if (Q->n != 0.0) {
         lp.phi = xy.y;
         for (i = MAX_ITER; i ; --i) {
-            lp.phi -= V = (Q->n * lp.phi + Q->n1 * sin(lp.phi) - xy.y ) /
-                (Q->n + Q->n1 * cos(lp.phi));
+            const double V = (Q->n * lp.phi + Q->n1 * sin(lp.phi) - xy.y ) /
```

Comment:
Curious question: in cases like this, would a new address for `V` need to be created each iteration within this for-loop? Are there advantages of declaring the scope of the variable as `const double V` within the for-loop rather than with `double V` outside the for-loop?

Reply:
Restricting to the needed scope is a good practice for code clarity. This is the job of the compiler to map C/C++ variables to registers or stack where it sees fit. Trying to do that at hand by having a variable that is re-used for different purposes is generally just a bad idea, or a sequel of practices of older times where you had to provide hints such as the `register` keyword to help compilers.
Nowadays compilers are way smarter than humans for such decisions, and reducing the scope should actually help them. Actually, their smartness includes detecting that the V in the for loop is actually independent of the V after it :-) I've just verified that the assembly code in that function before and after my change is the same (at -02)
So the purpose of this pull request is mostly code clarity. There should be hardly any change in machine code generation.

Reply:
Thanks for the explanation. I figured it would be the job for the compiler's optimiser to determine.